### ISSUE TITLE
Method parameter type hint does not appear in the docs

### ISSUE THREAD

**Issue by JaZo** [FriendsOfPHP/Sami#281]
Best explained by this screenshot:

![class-typehint-issue](https://cloud.githubusercontent.com/assets/3475007/26304429/9407232e-3eeb-11e7-8521-b7ba6ad98e63.png)
The function getAmountByCurrency misses the type hint.

**PHP**
```php
namespace App\Collections;

use Illuminate\Support\Collection;
use Money\Currency;

class MoneyCollection extends Collection
{
    /**
     * @var \Money\Money[]
     */
    protected $items = [];

    /**
     * Grab's the correct amount by currency
     *
     * @param \Money\Currency $currency
     *
     * @return int|void
     */
    public function getAmountByCurrency(Currency $currency)
    {
        //
    }

    /**
     * Add a Money or MoneyCollection to the current collection, summing all existing currencies.
     *
     * @param \Money\Money|\App\Collections\MoneyCollection $value
     *
     * @return static
     */
    public function add($value)
    {
        //
    }
}
```

**Config**
```php
use Sami\Sami;
use Symfony\Component\Finder\Finder;

$iterator = Finder::create()
    ->files()
    ->name('*.php')
    ->in(__DIR__.'/app');

return new Sami(
    $iterator, [
        'theme'     => 'default',
        'build_dir' => __DIR__.'/docs/sami',
        'cache_dir' => __DIR__.'/storage/sami/cache',
    ]
);
```

**Sami version**
4.0.3

**Comment by aik099** [comment]
What words are missing and in which place in documents. By comparing docs & code I just don't see them.

Also what is class type hint?

**Comment by JaZo** [comment]
The problem occurs when documenting this function:
```php
/**
 * Grab's the correct amount by currency
 *
 * @param \Money\Currency $currency
 *
 * @return int|void
 */
public function getAmountByCurrency(Currency $currency)
```

Expected output:
```
int|void getAmountByCurrency(Currency $currency)
```

Actual output:
```
int|void getAmountByCurrency( $currency)
```

N.B. With type hint I mean type declaration.

**Comment by JaZo** [comment]
Thanks, I corrected the title!

**Comment by TiS** [comment]
This is probably due to changes in `NodeVisitor::addMethod()` function. There are no unit tests for it, so it's hard to tell for sure.

**Comment by JaZo** [comment]
It looks like the fix breaks parameter typehints from PHPdocs. The problematic example function (getAmountByCurrency) is fixed, but the other function now misses typehints. Using the same input and config, but with 4.0.5 gives the following result:

### Correct
Input:
```php
/**
 * Grab's the correct amount by currency
 *
 * @param \Money\Currency $currency
 *
 * @return int|void
 */
public function getAmountByCurrency(Currency $currency)
```

Output:
```
int|void getAmountByCurrency(Currency $currency)
```

### Incorrect
Input:
```php
/**
 * Add a Money or MoneyCollection to the current collection, summing all existing currencies.
 *
 * @param \Money\Money|\App\Collections\MoneyCollection $value
 *
 * @return static
 */
public function add($value)
```

Expected output:
```
MoneyCollection add(Money|MoneyCollection $value)
```

Actual output:
```
MoneyCollection add( $value)
````

**Comment by TiS** [comment]
As I'm responsible for recent break, I'll try to add this case to tests and submit another PR in one or two days.

**Comment by JaZo** [comment]
Thanks for the fix, 4.0.6 works as expected!
